server/client.go

```diff
@@ -1215,41 +1279,95 @@ func (c *client) processSub(argo []byte) (err error) {
 		return nil
 	}
 
+	// Check if we have a maximum on the number of subscriptions.
 	if c.msubs > 0 && len(c.subs) >= c.msubs {
 		c.mu.Unlock()
 		c.maxSubsExceeded()
 		return nil
 	}
 
-	// Check if we have a maximum on the number of subscriptions.
 	// We can have two SUB protocols coming from a route due to some
 	// race conditions. We should make sure that we process only one.
 	sid := string(sub.sid)
+	var chkImports bool
+
 	if c.subs[sid] == nil {
 		c.subs[sid] = sub
-		if c.srv != nil {
-			err = c.srv.sl.Insert(sub)
+		if c.sl != nil {
+			err = c.sl.Insert(sub)
 			if err != nil {
 				delete(c.subs, sid)
 			} else {
+				if c.acc != nil {
+					chkImports = true
+				}
 				shouldForward = c.typ != ROUTER
 			}
 		}
 	}
 	c.mu.Unlock()
+
 	if err != nil {
 		c.sendErr("Invalid Subject")
 		return nil
 	} else if c.opts.Verbose {
 		c.sendOK()
 	}
+	if chkImports {
+		if err := c.checkAccountImports(sub); err != nil {
+			c.Errorf(err.Error())
+		}
+	}
 	if shouldForward {
 		c.srv.broadcastSubscribe(sub)
 	}
 
 	return nil
 }
 
+// Check to see if we need to create a shadow subscription due to imports
+// in other accounts.
+// Assume lock is held
```

Comment:
Actually not, this function grabs the lock.

Reply:
fixed.